### PR TITLE
Small clean up and utils reorg

### PR DESCRIPTION
**Pull request by 0sewa0**
This PR implements a new data enhancement to...

## Categories

- [ ] Bugfix
- [ ] Enhancement
- [ ] Backporting
- [x] Others (CI, Infrastructure, Documentation)

## Sample archive

- not necessary

## Documentation

- not necessary

## Unit Tests

- not necessary

## Privacy

Yes. There are no sensitive data in the newly collected information.

### CODE REVIEW

File: pkg/gather/clusterconfig/0_gatherer_test.go [hunk 1]
```diff
@@ -17,14 +18,14 @@ func (e *testError) Error() string {
 func mockGatherFunction1(g *Gatherer, c chan<- gatherResult) {
 	c <- gatherResult{[]record.Record{{
 		Name: "config/mock1",
-		Item: Raw{"mock1"},
+		Item: marshal.Raw{Str: "mock1"},
```
Comment (rluders) [inline pkg/gather/clusterconfig/0_gatherer_test.go:21]: Isn't this one duplicated with [this](https://github.com/openshift/insights-operator/blob/dcca69ec2d24e9efa045586253b5354049e96ee3/tests/reports.go)? Is this marshal only used for testing? If the answer is YES, I'll recommend moving it to the `/tests/` packages. Same for everything else that is only used as a testing toolkit.

**Review by rluders — CHANGES_REQUESTED** [review]

File: pkg/gather/clusterconfig/0_gatherer_test.go (hunk 1 above)
Comment (0sewa0) [reply pkg/gather/clusterconfig/0_gatherer_test.go:21]: its not only used for testing,
 its used in `pkg/gather/clusterconfig/gather_logs.go`, `pkg/gather/clusterconfig/operators.go` and pkg/gather/clusterconfig/version.go

Comment (rluders) [reply pkg/gather/clusterconfig/0_gatherer_test.go:21]: Gotcha.

**Comment by rluders** [comment]
Tested locally.
/lgtm

**Pull request merged**